File: 1086-divisor-game/divisor-game.js
/**
 * @param {number} n
 * @return {boolean}
 */

function recursive(n,alice){
    if(n===0){
        return;
    }

    for(let i=1;i<n;i++){
        alice[0]=!alice[0];
        
        recursive(n-i,alice);
        break;
    }
}
var divisorGame = function(n) {
    let alice=[0,1];
    recursive(n,alice);
    return alice[0];
};

File: 1086-divisor-game/divisor-game.cpp
class Solution {
public:
   void solveUsingRecursion(int n,bool &alice){
    if(n==0){
        return;
    }
    for(int i=1;i<n;i++)
    {
        if(n%i==0){
            alice=!alice;
            solveUsingRecursion(n-i,alice);
            break;
        }
    }

    

   }
    bool divisorGame(int n) {
        bool alice=false;
        solveUsingRecursion(n,alice);
        return alice;
        
    }
};